metricbeat-tests/runner_test.go

```diff
@@ -54,78 +58,85 @@ func TestMain(m *testing.M) {
 	os.Exit(status)
 }
 
-func cleanUpOutputs() {
-	dir, _ := os.Getwd()
-
-	pattern := dir + "/outputs/*.metrics*"
-
-	files, err := filepath.Glob(pattern)
-	if err != nil {
-		log.WithFields(log.Fields{
-			"error": err,
-			"path":  pattern,
-		}).Fatal("Cannot retrieve outputs :(")
+// assertHitsArePresent returns an error if no hits are present
+func assertHitsArePresent(hits map[string]interface{}, q ElasticsearchQuery) error {
+	hitsCount := len(hits["hits"].(map[string]interface{})["hits"].([]interface{}))
+	if hitsCount == 0 {
+		return fmt.Errorf(
+			"There aren't documents for %s-%s on Metricbeat index",
+			q.EventModule, q.ServiceVersion)
 	}
 
-	for _, f := range files {
-		err := os.Remove(f)
-		if err != nil {
-			log.WithFields(log.Fields{
-				"error": err,
-				"path":  f,
-			}).Fatal("Cannot remove output file :(")
-		}
-	}
+	return nil
 }
 
-func metricbeatOutputsMetricsToTheFile(fileName string) error {
-	dir, _ := os.Getwd()
-
-	filePath := dir + "/outputs/" + fileName
-
-	foundChannel := make(chan bool, 1)
-
-	go fileChecker(foundChannel)
-	initialLoops := 15
-	loops := initialLoops
-	seconds := 2
-
-	for {
-		exists := fileExists(filePath)
-		if exists {
-			foundChannel <- true
-			close(foundChannel)
-			break
+// assertHitsDoNotContainErrors returns an error if any of the returned entries contains
+// an "error.message" field in the "_source" document
+func assertHitsDoNotContainErrors(hits map[string]interface{}, q ElasticsearchQuery) error {
+	for _, hit := range hits["hits"].(map[string]interface{})["hits"].([]interface{}) {
+		source := hit.(map[string]interface{})["_source"]
+		if val, ok := source.(map[string]interface{})["error"]; ok {
+			if msg, exists := val.(map[string]interface{})["message"]; exists {
+				log.WithFields(log.Fields{
+					"ID":            hit.(map[string]interface{})["_id"],
+					"error.message": msg,
+				}).Error("Error Hit found")
+
+				return fmt.Errorf(
+					"There are errors for %s-%s on Metricbeat index",
+					q.EventModule, q.ServiceVersion)
+			}
 		}
+	}
 
-		if loops == 0 {
-			return fmt.Errorf("Could not find the file %s after %d seconds", fileName, (initialLoops * seconds))
-		}
+	return nil
+}
 
-		log.WithFields(log.Fields{
-			"file":    fileName,
-			"times":   loops,
-			"seconds": (loops * seconds),
-		}).Debug("Waiting for the file to be present")
+func thereAreNoErrorsInTheIndex(metricbeatVersion string) error {
+	esIndexName := strings.ReplaceAll(metricbeatVersion, "-SNAPSHOT", "")
+	now := time.Now()
+
+	formattedDate := strings.ReplaceAll(now.Format("2006-01-02"), "-", ".")
```

Comment:
This can introduce flakiness on midnight, what do you think about using wildcards and querying `metricbeat-*` instead of building a full index name?

Reply:
Yes, I mentioned in the PR, I don't like this hardcoded string either :)

Reply:
Using wildcards is an option to remove the ugly parts :slightly_smiling_face:

Reply:
Applied